store/postgres/migration.go

```diff
@@ -0,0 +1,185 @@
+package postgres
+
+import (
+	"context"
+	"embed"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/source/iofs"
+	"github.com/odpf/salt/log"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+
+	"github.com/odpf/optimus/store"
+)
+
+//go:embed migrations
+var migrationFs embed.FS
+
+type migrationStep struct {
+	CurrentOptimusVersion   string
+	CurrentMigrationVersion uint
+	PreviousOptimusVersion  string
+	CreatedAt               time.Time
+}
+
+type migration struct {
+	logger                 log.Logger
+	incomingOptimusVersion string
+
+	db      *gorm.DB
+	migrate *migrate.Migrate
+}
+
+// NewMigration initializes migration mechanism specific for postgres
+func NewMigration(logger log.Logger, incomingOptimusVersion, dbConnURL string) (store.Migration, error) {
+	if logger == nil {
+		return nil, errors.New("logger is nil")
+	}
+	if incomingOptimusVersion == "" {
+		return nil, errors.New("incoming optimus version is empty")
+	}
+	if dbConnURL == "" {
+		return nil, errors.New("database connection url is empty")
+	}
+	m, err := newMigrate(dbConnURL)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing migrate: %w", err)
+	}
+	db, err := gorm.Open(postgres.Open(dbConnURL))
```

Comment:
We are opening a separate connection for migrate, where are we closing this connection ?

Reply:
updated by adding explicit close statement